src/backend/joanie/tests/core/api/organizations/test_contracts_signature_link.py

```diff
@@ -0,0 +1,343 @@
+        """
+        When passing a list of contract ids,
+        only the contracts with these ids should be signed.
+        """
+
+        organization = factories.OrganizationFactory()
+        relation = factories.CourseProductRelationFactory(
+            organizations=[organization],
+            product__contract_definition=factories.ContractDefinitionFactory(),
+        )
+        orders = factories.OrderFactory.create_batch(
+            2,
+            product=relation.product,
+            course=relation.course,
+            organization=organization,
+            state=enums.ORDER_STATE_VALIDATED,
+            product__contract_definition=factories.ContractDefinitionFactory(),
+        )
+        access = factories.UserOrganizationAccessFactory(
+            organization=organization, role="owner"
+        )
+
+        for order in orders:
+            order.submit_for_signature(order.owner)
+            order.contract.submitted_for_signature_on = timezone.now()
+            order.contract.student_signed_on = timezone.now()
+
+        token = self.generate_token_from_user(access.user)
+
+        response = self.client.get(
+            f"/api/v1.0/organizations/{orders[0].organization.id}/contracts-signature-link/",
+            HTTP_AUTHORIZATION=f"Bearer {token}",
+            data={"contract_ids": [orders[0].contract.id]},
+        )
+
+        self.assertEqual(response.status_code, HTTPStatus.OK)
+        content = response.json()
+        self.assertIn(
+            "https://dummysignaturebackend.fr/?requestToken=",
+            content["invitation_link"],
+        )
+
+        self.assertCountEqual(content["contract_ids"], [str(orders[0].contract.id)])
+
+    def test_api_organization_contracts_signature_link_exclude_canceled_orders(self):
+        """
+        Authenticated users with owner role should be able to sign contracts in bulk but
+        not validated orders should be excluded.
+        """
+        order = factories.OrderFactory(
+            state=enums.ORDER_STATE_VALIDATED,
+            product__contract_definition=factories.ContractDefinitionFactory(),
+        )
+        access = factories.UserOrganizationAccessFactory(
+            organization=order.organization, role="owner"
+        )
+        order.submit_for_signature(order.owner)
+        order.contract.submitted_for_signature_on = timezone.now()
+        order.contract.student_signed_on = timezone.now()
+        # Simulate the usser has signed its contract then later canceled its order
+        order.cancel()
+
+        token = self.generate_token_from_user(access.user)
+
+        response = self.client.get(
+            f"/api/v1.0/organizations/{order.organization.id}/contracts-signature-link/",
+            HTTP_AUTHORIZATION=f"Bearer {token}",
+        )
+
+        self.assertEqual(response.status_code, HTTPStatus.BAD_REQUEST)
+        assert response.json() == {
+            "detail": "No contract to sign for this organization."
+        }
```

Comment:
```suggestion
        self.assertContains(
            response,
             "No contract to sign for this organization.",
             status_code=HTTPStatus.BAD_REQUEST
        )
```
Just a little comment, I also agree with your implementation.